File: Project Milestone 2/Devorah Jane/kafka/topic.js
//const Kafka = require("kafkajs").Kafka
const {Kafka} = require("kafkajs")

run();

async function run(){
    try{
        const kafka = new Kafka({
            "clientID": "myapp",
            "brokers": [
                "localhost:9093",
                "localhost:9094",
                "localhost:9095"
            ]
        })

        const admin = kafka.admin()
        console.log("Connecting....")
        await admin.connect()
        console.log("Connected....")

        //A-M, N-Z
        await admin.createTopics({  // Create topics
            "topics":[{
                "topic": "Users",   //name
                "numPartitions": 2  //num of partition
            }]
        })

        console.log("Created successfully!")
        await admin.disconnect();

    }
    catch(ex){
        console.error(`Something bad happened ${ex}`)
    }
    finally{
        process.exit
    }
}

File: Project Milestone 2/Esam/kafka-python/topic.py
from kafka.admin import KafkaAdminClient, NewTopic

topics = []

admin = KafkaAdminClient(
    bootstrap_servers=['localhost:9093','localhost:9094','localhost:9095'], 
    client_id='myapp'
)

topics.append(
    NewTopic(
        name="topic_1", 
        num_partitions=3, 
        replication_factor=3
        )
    )

admin.create_topics(
    new_topics=topics
    )

print("Topic created successfully!")